### DISCUSSION TITLE
Close client connection, but do the chores before service the next request

### DISCUSSION THREAD

**Discussion by lezhnev74** [roadrunner-server/roadrunner#1062] [Q&A] [answered]
Is it possible with the road runner to achieve this?
1. A worker accepts a new request.
2. The worker serves the request and sends back a response (like `fastcgi_finish_request`).
3. At this point the client received their response and probably closed the connection.
4. The worker continues doing some chores, including maybe cleaning up, calling the garbage collection etc.
5. When it is done the worked is ready to accept the next request.

Thanks! :pray:

**Comment by lezhnev74** [comment, marked as answer]
Ok, I guess this loop is exactly for that purpose. 

```
while (true) {
    $request = $psr7->waitRequest();
    $psr7->respond(new Psr7\Response(200, [], 'Hello RoadRunner!'));

   // clean up here <---------------
}
```

I am thinking how the client knows that there is nothing else to receive? Will `$psr7->respond(...)` be enough and client won't exxpect any extra payload?

**Reply by rustatian** [reply]
Hey @lezhnev74 You may use `try-catch-finally` inside the `while` loop and do the clean-up in the `finally` block. (I'm not a PHP programmer, but I guess that's should be possible).
You may also use the `max_jobs` option to specify, how many requests should handle worker before restarting (like a classic dying after request model).
Feel free to join our discord server: https://discord.gg/TFeEmCs

**Reply by rustatian** [reply]
> I am thinking how the client knows that there is nothing else to receive?

That is how HTTP or FCGI works. Request-response. Like, you need to make 3 requests, then you received 3 responses. So, now you know, that there is nothing to receive 😃
